lib/proxy/router.go

```diff
@@ -259,6 +271,37 @@ func (r *Router) DialHost(ctx context.Context, clientSrcAddr, clientDstAddr net.
 		case serverAddr == "" && target.GetUseTunnel():
 			serverAddr = reversetunnelclient.LocalNode
 		}
+		// If the node is a registered openssh node don't set agentGetter
+		// so a SSH user agent will not be created when connecting to the remote node.
+		if target.IsOpenSSHNode() {
+			agentGetter = nil
+			isAgentlessNode = true
+
+			// If the node is of SubKindOpenSSHNode, create the signer.
+			//
+			// Creating the ssh.Signer for SubKindOpenSSHEICENode requires the SSH Login User.
+			// SSH Login User is not available at this stage.
+			// So, the ssh.Signer is created later in the process, after the connection was set up and identity was received.
+			if target.GetSubKind() == types.SubKindOpenSSHNode {
+				client, err := r.GetSiteClient(ctx, clusterName)
+				if err != nil {
+					return nil, trace.Wrap(err)
+				}
+				sshSigner, err = signer(ctx, client)
+				if err != nil {
+					return nil, trace.Wrap(err)
+				}
+			}
+
+			if target.GetSubKind() == types.SubKindOpenSSHEICENode {
+				destinationListener, err = net.Listen("tcp", ":0")
```

Comment:
This looks insecure: we are listening on all interfaces here. Can we lock down this at least to localhost, or even better, make it completely in-process, making it only usable for in-process routing?

Reply:
You would still need the private key (which stays in memory) to actually access the target EC2 host.

>  making it only usable for in-process routing

How would we do this? The closest I can think of is a unix socket, but I'm not sure it has any advantage vs a `tcp,localhost:0` socket.
Playing with file permissions here seems over-engineering
What do you think?

I'll change it to `localhost:0`, if the above is not possible

Reply:
Ah, I missed that the private key will be needed for connection. Do you think we can add some docs or a test to cover that? I feel this is important detail.

Given that security measure, I think `localhost:0` is just fine.